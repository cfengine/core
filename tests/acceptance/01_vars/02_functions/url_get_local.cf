###########################################################
#
# Test url_get() locally
#
###########################################################

body common control
{
    inputs => { "../../default.cf.sub" };
    bundlesequence => { default($(this.promise_filename)) };
    version => "1.0";
}

###########################################################

bundle agent init
{
  vars:
      "options_limit" data => '
{
  "url.max_content": 200,
  "nosuchoption": 100,
  "url.verbose": 0,
  "url.headers": [ "Foo: bar" ]
}';

      "options_clean" data => '{}';
      # local file: does it work on Windows?
      "res4" data => url_get("file://$(this.promise_filename)", options_limit);
}

bundle agent test
{
  meta:
      "test_skip_unsupported" string => "!feature_curl";

  vars:
      "kept" data => mergedata(
                               '{ "res4": init.res4[content] }',
                               '{ "res4/returncode": init.res4[returncode] }',
                               '{ "res4/rc": init.res4[rc] }',
                               '{ "res4/success": init.res4[success] }'
      );
}

###########################################################

bundle agent check
{
  methods:
      "check"  usebundle => dcs_check_state(test,
                                           "$(this.promise_filename).expected.json",
                                           $(this.promise_filename));
}
