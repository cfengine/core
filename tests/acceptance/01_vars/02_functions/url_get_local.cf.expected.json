{
  "kept": {
    "res4": "###########################################################\n#\n# Test url_get() locally\n#\n###########################################################\n\nbody common control\n{\n    inputs => { \"../../defau",
    "res4/rc": 0,
    "res4/returncode": 0,
    "res4/success": true
  }
}
